client/core/trade.go

```diff
@@ -1397,35 +1474,38 @@ func (t *trackedTrade) shouldBeginFindRedemption(ctx context.Context, match *mat
 	if spent { // NOTE: spent may not be accurate for SPV wallet, so this should not be a requirement.
 		t.dc.log.Infof("Swap contract for revoked match %s, order %s is spent. Will begin search for redemption", match, t.ID())
 	}
-	return confs >= t.wallets.fromAsset.SwapConf || spent
+	return spent || confs >= t.wallets.fromAsset.SwapConf
 }
 
 // tick will check for and perform any match actions necessary.
 func (c *Core) tick(t *trackedTrade) (assetMap, error) {
+	assets := make(assetMap) // callers expect non-nil map even on error :(
+
+	if !atomic.CompareAndSwapUint32(&t.ticking, 0, 1) {
+		// Another tick is running for this trade. Return so we don't block on
+		// t.mtx.Lock, and don't run concurrent checks since the results may
+		// become inaccurate when/if the other goroutine begins acting, and we
+		// MUST NOT take the same action twice.
+		c.log.Tracef("Skipping concurrent tick for trade %v", t.ID())
+		return assets, nil
+	}
+	defer atomic.StoreUint32(&t.ticking, 0)
```

Comment:
Right.  That occurred to me too.

I'm not sure how else to handle this though, because with out an all encompassing write lock, we simply cannot allow concurrent ticks to be making the same list of actions to take.

Reply:
Maybe a tick *mutex* that is just for this purpose.

Reply:
Gonna try this instead:

```go
// tick will check for and perform any match actions necessary.
func (c *Core) tick(t *trackedTrade) (assetMap, error) {
	assets := make(assetMap) // callers expect non-nil map even on error :(

	// Another tick maybe running for this trade. We have a mutext just for this
	// so we don't have to write-lock t.mtx.Lock, which would block many other
	// actions such as isActive. We can't just run concurrent checks since the
	// results may become inaccurate when/if the other goroutine begins acting,
	// and we MUST NOT take the same action twice.
	t.tickLock.Lock()
	defer t.tickLock.Unlock()
```